sonar-text-plugin/src/main/java/org/sonar/plugins/common/Analyzer.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * SonarQube Text Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.common;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonarsource.analyzer.commons.ProgressReport;
+
+public class Analyzer {
+  private static final int REPORT_REFRESH_TIME_IN_SECONDS = 10;
+  private static final Logger LOG = LoggerFactory.getLogger(Analyzer.class);
+  private boolean displayHelpAboutExcludingBinaryFile = true;
+
+  private SensorContext sensorContext;
+  private List<Check> activeChecks;
+  private NotBinaryFilePredicate notBinaryFilePredicate;
+  private boolean analyzeAllFilesMode;
+
+  public Analyzer() {
+    // Empty analyzer by default, configuration provided by setters
+  }
```

Comment:
Looks more like a builder pattern should be used.

Reply:
But I'm not sure this is necessary. The analyzer mainly depends on `sensorContext` and the files. Why don't we give everything at once to a static method of the Analyzer without constructing the Analyzer?

Reply:
Mmhh indeed, the builder logic doesn't look really necessary in the end.
It made more sense in a previous version where I was iterating manually and calling the analyzer on each file, but now this logic was also moved in the `Analyzer` class. Now it's about instantiating a single object to then call a single method, doesn't really make sense.